webcomponents/core/src/components/deck/deckdeckgo-deck/deckdeckgo-deck.tsx

```diff
@@ -84,10 +84,13 @@ export class DeckdeckgoDeck {
   @Prop({reflect: true}) direction: 'horizontal' | 'vertical' | 'papyrus' = 'horizontal';
   @Prop({reflect: true}) directionMobile: 'horizontal' | 'vertical' | 'papyrus' = 'papyrus';
 
+  @Prop({reflect: true}) autoSlide: boolean = false;
+
   @State()
   private dir: 'horizontal' | 'vertical' | 'papyrus';
 
   private observer: IntersectionObserver;
+  private slideLoopInterval: NodeJS.Timeout;
```

Comment:
Which version of node are you using? When I build the  component I get `Cannot find namespace 'NodeJS'` but that isn't the first time.  For me setInterval  returns  a number.

Not really  linked  to the PR, more something I have to clear...

Reply:
Tested on node: 12.11.1
which version do you have? I'll give it a look also in your version

Reply:
I've got  v14.8.0

I ask myself if  it's question of local version of if somehow I should enforce it in the component, but that's more a general question, not related to this  component/PR. I think I had such conflicts already. Definitely not a blocker here, more something to answer...

Reply:
Well, i had an issue with the 

```
this.idleMouseTimer = setTimeout(async () => {
  await this.showHideMouseCursor(false);
}, 2000);
```

Had to wrap the setTimeout with Number to silent the error
`idleMouseTimer: number;` while setTimeout returns NodeJS.Timeout and not number

I think is more with the current typescript version rather than Node
My typescript version is 3.8.3

Reply:
Maybe, Stencil uses Typescript v4.

Looks, if it's ok for you, we let it like that and once merged I take care of settings this type such as it works on my machine, ok?

Reply:
Sure, no problem!